bakery/src/scripts/link-book.py

```diff
@@ -0,0 +1,61 @@
+"""
+Replaces legacy module ids in links to external modules with
+uuids from the target module and corresponding canonical book.
+"""
+
+import sys
+import os
+import json
+from lxml import etree
+import requests
+
+
+def main():
+    data_dir, canonical_list = sys.argv[1:3]
+
+    # define the canonical books
+    with open(canonical_list) as canonical_file:
+        canonical_books = json.load(canonical_file)["canonical_books"]
+        canonical_ids = [book["uuid"] for book in canonical_books]
+
+    # load assembled collection
+    assembled_collection = f"{data_dir}/collection.assembled.xhtml"
+    doc = etree.parse(assembled_collection)
+
+    # look up uuids for external module links
+    for node in doc.xpath(
+        '//x:a[@href and starts-with(@href, "/contents/m")]',
+        namespaces={"x": "http://www.w3.org/1999/xhtml"},
+    ):
+        # find legacy module id
+        link = node.attrib["href"]
+        module_id = link.lstrip("/contents/")
+
+        # get target module uuid
+        req = requests.get(f"https://archive.cnx.org/content/{module_id}")
+        if req.status_code != 200:
+            continue
+        module_uuid = req.url.split("/")[-1]
+
+        # get list of books containing module
+        req = requests.get(f"https://archive.cnx.org/extras/{module_uuid}")
+        content = json.loads(req.content)
+        containing_books = {
+            book["ident_hash"].split("@")[0]: book["title"] for book in content["books"]
+        }
+
+        # match uuid in canonical book list
+        for book_uuid in canonical_ids:
+            if book_uuid not in containing_books:
+                continue
+            node.attrib["href"] = f"/contents/{module_uuid}"
+            node.attrib["data-book-uuid"] = book_uuid
+            break
```

Comment:
I can't find it anywhere now in the swarm doc notes, but I believe there was an error case we wanted to watch for in this script (see related notes in [cnx 1026](https://github.com/openstax/cnx/issues/1026) though I don't think they reflect the most recent convos on this point). 

IIRC, I think we ended up with wanting to error when there are multiple books returned from extras AND none of them are in our canonical list. The idea being if there is only one book returned from extras and it's not in our list it won't matter, but otherwise if we don't get a hit we should fail loudly. Does that ring a bell / sound right to you? Currently it looks like we won't modify the link if there's no hit in the list (?)